CommonTools/UtilAlgos/interface/FwdPtrProducer.h

```diff
@@ -46,6 +47,15 @@ namespace edm {
       iEvent.put(std::move(pOutputFwdPtr));
     }
 
+    static void fillDescriptions(edm::ConfigurationDescriptions& desc) {
+      edm::ParameterSetDescription ps;
+      if (std::is_same<reco::PFCandidate, T>::value)
+        ps.add<edm::InputTag>("src", {"particleFlowTmp"});
+      else
+        ps.add<edm::InputTag>("src");
```

Comment:
Please put a default also here, even if empty (ie, "").
ConfDB parsing requires a value, otherwise the parameter will not exist in ConfDb where we keep the HLT menus!

Reply:
I don't quite understand why we need to make `reco::PFCandidate` a special case. I would just supply a generic empty string in all cases.

Reply:
> Thought we wanted to have one fill description specifically for the PF case

This came into the context of a specific template specification involving PF because it's the only module of this type used in the HLT (which cares the most about this feature), but I really don't see the point of not making it general as you are touching the code.